java/flight/flight-jdbc-driver/src/main/java/org/apache/arrow/driver/jdbc/accessor/impl/calendar/ArrowFlightJdbcIntervalVectorAccessor.java

```diff
@@ -61,8 +54,14 @@ public ArrowFlightJdbcIntervalVectorAccessor(IntervalDayVector vector,
                                                ArrowFlightJdbcAccessorFactory.WasNullConsumer setCursorWasNull) {
     super(currentRowSupplier, setCursorWasNull);
     this.vector = vector;
-    this.stringBuilderGetter = vector::getAsStringBuilder;
-    this.objectClass = Duration.class;
+    stringGetter = (index) -> {
+      final NullableIntervalDayHolder holder = new NullableIntervalDayHolder();
+      vector.get(index, holder);
+      final int days = holder.days;
```

Comment:
If holder indicates the value is null, we can return null directly

Reply:
Done.